File: pset1/mario/more/mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    int height;
    // get height from user
    do
    {
        printf("Height: ");
        height = get_int();
    }
    while (!(height <= 23 && height >= 0));
    // print the pyramid
    for (int i = 1; i <= height; i++)
    {
        // print the first half of the pyramid
        for (int k = 0; k < height - i; k++)
        {
            printf(" ");
        }
        for (int k = 0; k < i; k++)
        {
            printf("#");
        }
        // print the space in the middle of the pyramid
        for (int k = 0; k < 2; k++)
        {
            printf(" ");
        }
        // print the second half of the pyramid
        for (int k = 0; k < i; k++)
        {
            printf("#");
        }
        // set the cursor to next line in the pyramid
        printf("\n");
    }
}

File: pset6/mario.py
# get input from user
while 1:
    height = int(input("Height: "))
    if height >= 0 and height <= 23:
        break
# print the pyramid
for i in range(1, height + 1):
    # the first half of the pyramid
    for k in range(0, height - i):
        print(" ", end="")
    for k in range(0, i):
        print("#", end="")
    # middle of the pyramid
    for k in range(0, 2):
        print(" ", end="")
    # the second half of pyramid
    for k in range(0, i):
        print("#", end="")
    # new level in pyramid
    print("")
